### DISCUSSION TITLE
Forcing certain tasks to run in serial

### DISCUSSION THREAD

**Discussion by multimeric** [ropensci/targets#930] [Help]
I have a large number of tasks to run using the futures executor, each of which appends to an SQLite database. Each of these append tasks needs to be separate so that `targets` can incrementally run them, keep track of which ones have succeeded and which have failed, and recover after a failure. However, SQLite doesn't like concurrent writes, so for this particular set of tasks, I want them to run in serial. How can I implement this?

A simplified example of my pipeline is this. Here the first task just creates a list of datasets to load, and the second task maps over the list of datasets and loads each individually (I forget if the `pattern` argument is correct, but hopefully you get the gist). This second task, `loaded_db` is the one I want to run in serial:
```R
list(
  tar_files(
    name = datasets,
    command = get_datasets()
  ),
  tar_target(
    name = loaded_db,
    format = "file",
    command = append_to_database(datasets),
    pattern = map(datasets)
  )
)
```

The only mechanism I am aware of for limiting concurrency is setting `tar_make_future(workers=1)`, but of course this will force *all* tasks to be run in serial, but actually I only want this to happen for a specific set of branches. 

This question is broadly a copy of https://stackoverflow.com/q/67809168/2148718, but I don't think that question received a particularly good answer.

One possible solution I can imagine is forcing a dependency between branches, so that the DAG goes from this:
```mermaid
flowchart LR
A --> B;
A --> C;
A --> D;
```

To this:

```mermaid
flowchart LR
A --> B;
B --> C;
C --> D;
```

However I'm not sure how to actually do this in targets.

**Comment by wlandau** [comment]
In your setup, you could set `deployment = "main"` in the `loaded_db` target to run all those branches sequentially. Those branches will run on the main controlling process rather than individual futures.

**Reply by multimeric** [reply]
That seems like a useful workaround. Although I guess it's not a general mechanism for specifying the concurrency.
